src/Microsoft.Net.Http.Headers/MediaTypeHeaderValue.cs

```diff
@@ -170,55 +205,141 @@ public IList<NameValueHeaderValue> Parameters
         {
             get
             {
-                return _mediaType.Subsegment(_mediaType.IndexOf('/') + 1);
+                return _mediaType.Subsegment(_mediaType.IndexOf(ForwardSlashCharacter) + 1);
+            }
+        }
+
+        /// <summary>
+        /// Gets the subtype of the <see cref="MediaTypeHeaderValue"/>, excluding any structured syntax suffix.
+        /// </summary>
+        /// <example>
+        /// For the media type <c>"application/vnd.example+json"</c>, the property gives the value
+        /// <c>"vnd.example"</c>.
+        /// </example>
+        public StringSegment SubTypeWithoutSuffix
+        {
+            get
+            {
+                var subTypeSuffixLength = _mediaType.LastIndexOf(PlusCharacter);
+                var length = _mediaType.IndexOf(ForwardSlashCharacter) + 1;
+                if (subTypeSuffixLength == -1)
+                {
+                    return _mediaType.Subsegment(length);
+                }
+                return _mediaType.Subsegment(length, subTypeSuffixLength - length);
             }
         }
 
         /// <summary>
-        /// MediaType = "*/*"
+        /// Gets the structured syntax suffix of the <see cref="MediaTypeHeaderValue"/> if it has one.
+        /// See <see href="https://tools.ietf.org/html/rfc6838#section-4.8">The RFC documentation on structured syntaxes.</see>
         /// </summary>
-        public bool MatchesAllTypes
+        /// <example>
+        /// For the media type <c>"application/vnd.example+json"</c>, the property gives the value
+        /// <c>"json"</c>.
+        /// </example>
+        public StringSegment SubTypeSuffix
         {
             get
             {
-                return MediaType.Equals("*/*", StringComparison.Ordinal);
+                var subTypeSuffixLength = _mediaType.LastIndexOf(PlusCharacter);
+                if (subTypeSuffixLength == -1)
+                {
+                    return default(StringSegment);
+                }
+                return _mediaType.Subsegment(subTypeSuffixLength + 1);
             }
         }
 
+
         /// <summary>
-        /// SubType = "*"
+        /// Get a <see cref="IList{T}"/> of facets of the <see cref="MediaTypeHeaderValue"/>. Facets are a
+        /// period separated list of StringSegments in the <see cref="SubTypeWithoutSuffix"/>.
+        /// See <see href="https://tools.ietf.org/html/rfc6838#section-3">The RFC documentation on facets.</see>
         /// </summary>
-        public bool MatchesAllSubTypes
+        /// <example>
+        /// For the media type <c>"application/vnd.example+json"</c>, the property gives the value:
+        /// <c>{"vnd", "example"}</c>
+        /// </example>
+        public IList<StringSegment> Facets
         {
             get
             {
-                return SubType.Equals("*", StringComparison.Ordinal);
+                return SubTypeWithoutSuffix.Split(new char[] { PeriodCharacter } ).ToList();
```

Comment:
The way MTHV is design is to do the parsing on get, rather than on set. I'm assuming you are just referring to caching the new char[] { PeriodCharacter } though.